gusto/recovery/recovery_options.py

```diff
@@ -0,0 +1,104 @@
+
+from firedrake import (interval, FiniteElement, TensorProductElement, FunctionSpace,
+                       VectorFunctionSpace)
+from gusto.core.function_spaces import DeRhamComplex
+from gusto.core.configuration import RecoveryOptions
+
+
+class RecoverySpaces(object):
+    """
+    Finds or builds necessary spaces to carry out recovery transport for lowest
+    and mixed order domains (0,0), (0,1) and  (1,0)
+    """
+
+    def __init__(self, domain, boundary_method=None, use_vector_spaces=False):
+        """
+        Args:
+            domain (:class:`Domain`): the model's domain object, containing the
+            mesh and the compatible function spaces.
+
+            boundary_method (:variable:'dict', optional: A dictionary containing the space
+            the boundary method is to be applied to along with specified method. Acceptable keys are "DG",
+            "HDiv" and theta, defaults to None
+
+            use_vector_spaces (boolean, optional) Determins if we need to use DG / CG
```

Comment:
spelling: determines and Hcurl, `bool` rather than `boolean` and put a colon after the type - necessary for consistency but also because this can affect how the docs look